packages/hdwallet-walletconnect-bridge/src/bridge.ts

```diff
@@ -0,0 +1,149 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import WalletConnect from "@walletconnect/client";
+import { IWalletConnectSession } from "@walletconnect/types";
+import { convertHexToUtf8 } from "@walletconnect/utils";
+
+import { WalletConnectCallRequest, WalletConnectSessionRequestPayload } from "./types";
+
+const addressNList = core.bip32ToAddressNList("m/44'/60'/0'/0/0");
+
+export class HDWalletWCBridge {
+  constructor(private readonly wallet: core.ETHWallet, private readonly connector: WalletConnect) {}
+
+  static fromURI(uri: string, wallet: core.ETHWallet) {
+    return new HDWalletWCBridge(wallet, new WalletConnect({ uri }));
+  }
+
+  static fromSession(session: IWalletConnectSession, wallet: core.ETHWallet) {
+    return new HDWalletWCBridge(wallet, new WalletConnect({ session }));
+  }
+
+  async connect() {
+    if (!this.wallet) throw new Error("Missing ETH Wallet to connect with");
+
+    if (!this.connector.connected) {
+      await this.connector.createSession();
+    }
+
+    this.subscribeToEvents();
+  }
+
+  async disconnect() {
+    await this.connector.killSession();
+    this.connector.off("session_request");
+    this.connector.off("session_update");
+    this.connector.off("connect");
+    this.connector.off("disconnect");
+    this.connector.off("call_request");
+  }
+
+  private subscribeToEvents() {
+    this.connector.on("session_request", this._onSessionRequest.bind(this));
+    this.connector.on("session_update", this._onSessionUpdate.bind(this));
+    this.connector.on("connect", this._onConnect.bind(this));
+    this.connector.on("disconnect", this._onDisonnect.bind(this));
+    this.connector.on("call_request", this._onCallRequest.bind(this));
+  }
+
+  async _onSessionRequest(error: Error | null, payload: WalletConnectSessionRequestPayload) {
+    this.log("Session Request", { error, payload });
+
+    const address = await this.wallet.ethGetAddress({ addressNList });
+    this.connector.approveSession({
+      chainId: payload.params[0].chainId ?? 4,
+      accounts: [address!],
```

Comment:
This is triggering the `@typescript-eslint/no-non-null-assertion` rule in the CI lint job